transcom/mymove
# B-21686: Allow Prime Sim User to update estimated weight and actual weight independently

## [Agility ticket](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-21686)

## Summary
Issue was that the Prime Sim User could not update Estimated Weight and Actual Weight Independently. This was happening because there is a default 0 in both text fields so if it is not removed then it would get submitted as 0, which would violate the validation rule of a value of 1 or greater. Prime API already supports updating both fields independently. The issue was the prime sim was submitting the payload with default value of 0.

Is there anything you would like reviewers to give additional scrutiny?

[this article](tbd) explains more about the approach used.

## Verification Steps for the Author

These are to be checked by the author.


## Verification Steps for Reviewers

These are to be checked by a reviewer.


### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/getting-started/application-setup/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/getting-started/development/testing)

### How to test

1. Access the Prime Sim
2. Start with a shipment that has no estimated weight nor an actual weight
3. Try updating actual weight with a value greater than 0. Save the changes. Should be successful.
4. To speed up testing so you do not have to create a new shipment null out the actual weight , then update the estimated weight on the shipment as the prime.
5. Try adding a 0 in the field and you will get the validation error.

### Frontend


### Backend


### Database

#### Any new migrations/schema changes:


## Screenshots
